ark/utils/deepcell_service_utils.py

```diff
@@ -0,0 +1,93 @@
+from twisted.internet import reactor
+from kiosk_client import manager
+import os
+import glob
+from zipfile import ZipFile
+import warnings
+
+
+def create_deepcell_output(deepcell_input_dir, points, deepcell_output_dir,
+                           suffix='_feature_0', host='https://deepcell.org', job_type='multiplex'):
+    """ Handles all of the necessary data manipulation for running deepcell tasks.
+
+        Creates .zip files (to be used as input for DeepCell),
+        calls run_deepcell_task method,
+        and extracts zipped output files to the specified output location
+
+        Args:
+            points (list):
+                List of points in preprocessing pipeline
+            deepcell_input_dir (str):
+                Location of preprocessed files (assume deepcell_input_dir contains <point>.tif
+                for each point in points list)
+            suffix (str):
+                Suffix for DeepCell output filename. e.g. for pointX, DeepCell output
+                should be <pointX>+suffix.tif. Default: '_feature_0'
+            deepcell_output_dir (str):
+                Location to save DeepCell output (as .tif)
+            host (str):
+                Hostname and port for the kiosk-frontend API server
+                Default: 'https://deepcell.org'
+            job_type (str):
+                Name of job workflow (multiplex, segmentation, tracking)
+                Default: 'multiplex'
+        Raises:
+            ValueError:
+                Raised if there is some point X (from points list) s.t.
+                the file <deepcell_input_dir>/PointX.tif does not exist
+        """
+
+    zip_path = os.path.join(deepcell_input_dir, 'points.zip')
+    if os.path.isfile(zip_path):
+        warnings.warn(f'{zip_path} will be overwritten.')
+
+    with ZipFile(zip_path, 'w') as zipObj:
+        for point in points:
+            filename = os.path.join(deepcell_input_dir, point + '.tif')
+            if not os.path.isfile(filename):
+                raise ValueError('Could not find .tif file for %s. '
+                                 'Invalid value for %s' % (point, filename))
+            zipObj.write(filename, os.path.basename(filename))
+
+    run_deepcell_task(zip_path, deepcell_output_dir, host, job_type)
+    os.remove(zip_path)
+
+    # extract the .tif output
+    zip_files = glob.glob(os.path.join(deepcell_output_dir, '*.zip'))
+    zip_files.sort(key=os.path.getmtime)
+    with ZipFile(zip_files[-1], 'r') as zipObj:
+        zipObj.extractall(deepcell_output_dir)
+        for point in points:
+            if point + suffix + '.tif' not in zipObj.namelist():
+                warnings.warn(f'Deep Cell output file was not found for {point}.')
+
+
+def run_deepcell_task(input_dir, output_dir, host='https://deepcell.org',
```

Comment:
To clarify, we don't need a test function for this one?

Reply:
I couldn't think of any testing that would actually be useful. We'd have to mock basically the entire function, and at that point I'm not sure what we're actually testing. Open to suggestions if you think there's functionality worth testing

Reply:
Ok, I didn't think there was any additional testing we needed for this since its basically a driver function. Don't think we need to add a test function for this.